scripts/app.js

```diff
@@ -0,0 +1,119 @@
+const bookArray = []; // global array
+
+// MVP part
+/**
+ * This is the booj object
+ * @param {string} title The title of the book
+ * @param {string} author The author of the book
+ * @param {int} numPages The number of pages of the book
+ * @param {boolean}isRead Has the book been readed yet ?
+ */
+function Book(title, author, numPages, isRead) {
+  this.title = title;
+  this.author = author;
+  this.numPages = numPages;
+  this.isRead = isRead;
+}
+/**
+*@param{Object} book adds a newly created book into array
+*/
+function addBookToLibrary(book) {
+  bookArray.push(book);
+}
+/**
+*@param{int} index is the index where the book is stored in array
+*/
+function deleteBook(index) {
+  bookArray.splice(index, 1);
+  render();
+}
+// Listeners
+document.getElementById('submitBookBtn').addEventListener('click', createNewBook);
+
+/**
+* The handler of the button 'Submit book'
+@param{e} e is the event
+*/
+function createNewBook(e) {
```

Comment:
Let's have a separate function of adding new book into the library that has nothing to do with the DOM. The function that we have now mixes DOM and library related logic.

```javascript
// this should be how your addBookToLibrary is structured:
function addBookToLibrary(libaryArr, title, author, pages, isRead) = {
  const book = new Book(title, author, pages, isRead);
  libraryArr.push(book);
  // optionally returns the new book
  return book;
}

// then we can call it like this:
const newBook = addBookToLibrary(bookArray, "Harry Potter", "JK Rowling", 250, true)
```

What this does is simply create a new book and push that new book into your app's library.

Reply:
Fixed